homeassistant/components/backblaze/__init__.py

```diff
@@ -0,0 +1,72 @@
+"""Integration for Backblaze B2 Cloud Storage."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+
+from b2sdk.v2 import AuthInfoCache, B2Api, Bucket, InMemoryAccountInfo
+from b2sdk.v2.exception import InvalidAuthToken, NonExistentBucket
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed, ConfigEntryNotReady
+
+from .const import (
+    CONF_APPLICATION_KEY,
+    CONF_APPLICATION_KEY_ID,
+    CONF_BUCKET,
+    DATA_BACKUP_AGENT_LISTENERS,
+)
+
+type BackblazeConfigEntry = ConfigEntry[BackblazeonfigEntryData]
+
+
+@dataclass(kw_only=True)
+class BackblazeonfigEntryData:
+    """Dataclass holding all config entry data for a Backblaze entry."""
+
+    api: B2Api
+    bucket: Bucket
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: BackblazeConfigEntry) -> bool:
+    """Set up Backblaze from a config entry."""
+
+    info = InMemoryAccountInfo()
+    backblaze = B2Api(info, cache=AuthInfoCache(info))
+    try:
+        await hass.async_add_executor_job(
+            backblaze.authorize_account,
+            "production",
+            entry.data[CONF_APPLICATION_KEY_ID],
+            entry.data[CONF_APPLICATION_KEY],
+        )
+        bucket = await hass.async_add_executor_job(
+            backblaze.get_bucket_by_id, entry.data[CONF_BUCKET]
+        )
```

Comment:
Wrap in a sync function and do this with 1 executor call.

Reply:
I did in other places, not sure why I've skipped this one. Will do 👍